lib/moon/components/as_compoent.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Moon.Components.AsComponent do
```

Comment:
Why?

Reply:
some polymorphix mechanism in the name of HTML validation ("a" tag cannot include checkbox or radio) and React API alignment.
do you like PolymorphicComponent name more?

Reply:
also will need the same behaviour for Button component.

Reply:
https://github.com/coingaming/moon/issues/448

I think AsComponent is aligned with Design System and should not be added.

Reply:
@margusp2heathmont  are we going to align the api with React version?

Reply:
also we cannot render checkbox or radio inside <a> tag. Got HTML validation errors.